### ISSUE TITLE
[App]: Sweep cannot correctly draw the line chart

### ISSUE THREAD

**Issue by guanyipengai** [wandb/wandb#6456]
### Current Behavior

I use wandb.log to track the loss and accuracy for each run. However, in the line chart, their starting points are different and there is no discernible pattern to the starting points. **But sometimes it does work correctly.**

### Steps To Reproduce

Here is my sweep page :https://wandb.ai/guanyipeng/DGN-wisconsin?workspace=user-guanyipeng

### Screenshots

<img width="767" alt="截屏2023-10-15 11 02 23" src="https://github.com/wandb/wandb/assets/110985478/5cfec796-053f-4c43-aaa7-9e06cd0f8f55">

### Environment

OS:
MACOS14.5.1
Browsers:
Safari & Chrome
Version:
 wandb==0.15.11

### Additional Context

These are snippets of my code.
main.py
```python
    parser = argparse.ArgumentParser()
    parser.add_argument("--sweep", type=bool, default=True)
    parser.add_argument("--wandb_key", type=str, default=None)
    parser.add_argument('--device', type=str, default='cuda:0')
    parser.add_argument('--model', type=str, default='DGN')
    parser.add_argument('--dataset', type=str, default='wisconsin')
    parser.add_argument('--runs', type=int, default=10)
    parser.add_argument('--epoch', type=int, default=200)
    parser.add_argument('--node_dim', type=int, default=64)
    parser.add_argument('--num_channels', type=int, default=4)
    parser.add_argument('--lr', type=float, default=0.0003)
    parser.add_argument('--weight_decay', type=float, default=0.01)
    parser.add_argument('--num_layers', type=int, default=2)
    parser.add_argument("--channel_agg", type=str, default='concat')
    parser.add_argument("--remove_self_loops", action='store_true')
    args = parser.parse_args()
    pprint(args)

    if args.sweep:
        assert args.wandb_key is not None
        wandb.login(key=args.wandb_key)
        sweep_config = {'method': 'random'}
        metric = {'name': 'val_acc', 'goal': 'maximize'}
        sweep_config.update({'metric': metric})

        # Hyperparameters tuned by wandb
        parameters_dict = {'epoch': {'values': [200, 400, 600, 800]},
                           'node_dim': {'values': [32, 64, 128]},
                           'num_channels': {'values': [2,3,4,5,6,7,8]},
                           'lr': {
                               'values': [0.00001, 0.00003, 0.00006, 0.0001, 0.0003, 0.0006, 0.001, 0.003, 0.006, 0.01,
                                          0.03, 0.06, 0.1]},
                           'weight_decay': {
                               'values': [0.00001, 0.00003, 0.00006, 0.0001, 0.0003, 0.0006, 0.001, 0.003, 0.006, 0.01,
                                          0.03, 0.06]},
                           'num_layers': {'values': [1,2,3,4,5,6,7,8]}}

        # Hyperparameters that do not need adjustment using wandb
        parameters_dict.update({'model': {'value': args.model},
                                'dataset': {'value': args.dataset},
                                'runs': {'value': args.runs},
                                'device': {'value': args.device},
                                'channel_agg': {'value': args.channel_agg},
                                'remove_self_loops': {'value': args.remove_self_loops}})

        sweep_config.update({'parameters': parameters_dict})
        pprint(sweep_config)
        sweep_id = wandb.sweep(sweep_config, project='{}-{}-test-autodl'.format(args.model, args.dataset))
        wandb.agent(sweep_id, train, count=20)
    else:
        reproduce_paper_results(args)
``` 
train function:
```python
def train(config=None):
    init_seed(777)

    wandb.init(config=config)
    config = wandb.config

    device = torch.device(config.device)
    node_features, labels, edges = load_dataset(config.model, config.dataset)

    num_nodes = edges[0].shape[0]
    A = []
    for i, edge in enumerate(edges):
        edge_tmp = torch.from_numpy(np.vstack((edge.nonzero()[1], edge.nonzero()[0]))).to(device)
        value_tmp = torch.ones(edge_tmp.shape[1]).to(device)
        A.append((edge_tmp, value_tmp))
    edge_tmp = torch.stack((torch.arange(0, num_nodes), torch.arange(0, num_nodes))).to(device)
    value_tmp = torch.ones(num_nodes).to(device)
    A.append((edge_tmp, value_tmp))
    node_features = torch.from_numpy(node_features).to(device)

    tracker = {'train_loss': [], 'train_acc': [], 'val_loss': [], 'val_acc': [], 'test_loss': [], 'test_acc': [],
               'val_micro_f1': [], 'val_macro_f1': [],
               'test_micro_f1': [], 'test_macro_f1': []}

    for l in tqdm(range(config.runs)):

        train_node = torch.from_numpy(np.array(labels[l][0])[:, 0]).to(device)
        train_target = torch.from_numpy(np.array(labels[l][0])[:, 1]).to(device)
        valid_node = torch.from_numpy(np.array(labels[l][1])[:, 0]).to(device)
        valid_target = torch.from_numpy(np.array(labels[l][1])[:, 1]).to(device)
        test_node = torch.from_numpy(np.array(labels[l][2])[:, 0]).to(device)
        test_target = torch.from_numpy(np.array(labels[l][2])[:, 1]).to(device)
        num_classes = np.max(
            [torch.max(train_target).item(), torch.max(valid_target).item(),
             torch.max(test_target).item()]) + 1

        # initialize a model
        model = GTN(num_edge=len(A),
                    num_channels=config.num_channels,
                    w_in=node_features.shape[1],
                    w_out=config.node_dim,
                    num_class=num_classes,
                    num_layers=config.num_layers,
                    num_nodes=num_nodes,
                    args=config)

        optimizer = torch.optim.Adam(model.parameters(), lr=config.lr, weight_decay=config.weight_decay)

        model = model.to(device)

        for i in range(config.epoch):
            # print('Epoch ',i)
            model.zero_grad()
            model.train()
            loss, y_train, W = model(A, node_features, train_node, train_target)
            loss.backward()
            optimizer.step()

            train_acc = accuracy(torch.argmax(y_train.detach(), dim=1), train_target)
            tracker['train_acc'].append(train_acc)
            tracker['train_loss'].append(loss.detach().item())

            model.eval()
            # Valid
            with torch.no_grad():
                val_loss, y_valid, _ = model.forward(A, node_features, valid_node, valid_target)
                val_acc = accuracy(torch.argmax(y_valid, dim=1), valid_target)
                val_micro = sk_f1_score(valid_target.detach().cpu().numpy(), np.argmax(y_valid.cpu().numpy(), axis=1),
                                        average='micro')
                val_macro = sk_f1_score(valid_target.detach().cpu().numpy(), np.argmax(y_valid.cpu().numpy(), axis=1),
                                        average='macro')

                tracker['val_acc'].append(val_acc)
                tracker['val_loss'].append(val_loss.detach().item())
                tracker['val_micro_f1'].append(val_micro)
                tracker['val_macro_f1'].append(val_macro)

                test_loss, y_test, W = model.forward(A, node_features, test_node, test_target)
                test_acc = accuracy(torch.argmax(y_test, dim=1), test_target)
                test_micro = sk_f1_score(test_target.detach().cpu().numpy(), np.argmax(y_test.cpu().numpy(), axis=1),
                                         average='micro')
                test_macro = sk_f1_score(test_target.detach().cpu().numpy(), np.argmax(y_test.cpu().numpy(), axis=1),
                                         average='macro')

                tracker['test_acc'].append(test_acc)
                tracker['test_loss'].append(test_loss.detach().item())
                tracker['test_micro_f1'].append(test_micro)
                tracker['test_macro_f1'].append(test_macro)

    for key, item in tracker.items():
        tmp = [item[i:i + config.epoch] for i in range(0, len(item), config.epoch)]
        aver = np.mean(np.array(tmp), axis=0)

        for i in range(config.epoch):
            wandb.log({key: aver[i]})
        if key == 'test_acc':
            std = np.std(np.array(tmp), axis=0)
            for i in range(config.epoch):
                wandb.log({'test_acc_std': std[i]})

    wandb.finish()
```

**Comment by rsanandres-wandb** [comment]
Hello @guanyipengai !

Thank you for reporting this! I am looking into workspace and all the graphs seems to be centered correctly on my end. Are you still experiencing this error?

**Comment by guanyipengai** [comment]
> Hello @guanyipengai !
> 
> Thank you for reporting this! I am looking into workspace and all the graphs seems to be centered correctly on my end. Are you still experiencing this error?

Wow, all the pictures have now been correctly rendered. Perhaps this is related to the most recent update, but either way, thank you for providing such a great tool.
